### PR TITLE
Add the orphanet subset to the Slurp file

### LINKED ISSUE #527: Add the orphanet subset to the Slurp file
when reviewing the slurp/ordo.tsv, it would be helpful to know whether the orphanet term is in the "grouping disease", "disorder", or "subtype of disease" subset in orphanet. 
Please add a column in the slurp/ordo.tsv for the orphanet subset. Thank you!

### PR DESCRIPTION
**Pull request by joeflack4**
Resolves #527

## Overview
Adds rare disease subset information to `slurp/ordo.tsv` in a new column.

### Documentation
Was the documentation added/updated under `docs/`?
- [ ] Yes
- [x] No, updates to the docs were not necessary after careful consideration

### QC
Was the full pipeline run before submitting this PR using `sh run.sh make build-mondo-ingest` on this branch (after 
`docker pull obolibrary/odkfull:dev`), and no errors occurred?
- [ ] Yes
- [ ] No, there are no functional (code-related) changes to the pipeline in the PR, so no re-run is necessary
   
### New Packages
Were any new *Python packages* added?
- [ ] Yes, [requirements.txt.full](https://github.com/INCATools/ontology-development-kit/blob/master/requirements.txt.full) was updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [x] No

Were any other *non-Python packages* added?
- [ ] Yes, [Dockerfile](https://github.com/INCATools/ontology-development-kit/blob/master/Dockerfile) updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [x] No

### PR Review and Conversations Resolved
Has the PR been sufficiently reviewed by at least 1 team member of the Mondo Technical team and all threads resolved?
- [x] Yes

---

@souzadevinicius FYI

### CODE REVIEW

File: src/ontology/mondo-ingest.Makefile
Comment (twhetzel) [inline src/ontology/mondo-ingest.Makefile:622]: Is this getting the latest `mondo.sssom.tsv` file?

Comment (joeflack4) [reply src/ontology/mondo-ingest.Makefile]: ## This shouldn't have been part of the diff
@twhetzel Sorry, this is a wonky PR situation. That change is not part of this PR. It's part of #510, the branch for which this PR was used as a base.

I originally set this PR to be merged into `develop`, because that is where we do want to merge it. However, I forgot that this would make it so that all of the other changes in #510 would also show up here. I've now changed it so that this will merge into #510, so now the diff that shows is correct. The workflow we'll want to do, I think, is: (1) merge #510 into `develop` (we're just waiting on Nico's final explicit approval for that), and then I can (2) change the base for this to merge into `develop`.

Comment (twhetzel) [reply src/ontology/mondo-ingest.Makefile]: no worries. i just wanted to make sure that the new SOP/goal in https://github.com/monarch-initiative/mondo-ingest/issues/529 is accounted for here in `reports/ordo-subsets.robot.template.tsv` so that it also has the latest `mondo.sssom.tsv` file

Comment (joeflack4) [reply src/ontology/mondo-ingest.Makefile]: ## Answer to the question
In any case, as per your original question, there are several make goals which depend on `tmp/mondo.sssom.tsv`. You bring up a good point, related to #529. The way this dependency is set up, simply by listing `tmp/mondo.sssom.tsv` in the prerequisites of the goal, is the typical way to do it in `make`. However, in the event that the `mondo` repo has changed since `tmp/mondo.sssom.tsv` was generated, `make` will have no way of knowing that, and will think that `tmp/mondo.sssom.tsv` is up-to-date, and use the one that's presently there on the system.

Perhaps we're fine with that being the behavior. During a fresh build, this will not be a problem. It'll only be a problem in development situations. If we want to fix this for those situations as well, then, as far as I see it, we have 2 options.

Both options involve at least this change:
Move dependency into the goal's body. In this example, that would change it to:
```make
reports/ordo-subsets.robot.template.tsv: tmp/ordo-subsets.tsv 
	$(MAKE) tmp/mondo.sssom.tsv
	python3 $(SCRIPTSDIR)/ordo_subsets.py \
	...
```

### a. Always run a full refresh of `tmp/mondo.sssom.tsv`
Always run a refresh each time a goal is run which has it as a dependency.

This involves nothing else in addition to the change I just described above.

This could be be a real damper on local development. Not always terribly so, though. When I'm developing a goal that uses Python, I usually do all of my iteration in the debugger, and then run the make goal once at the end just to make sure it works.

### b. Intelligently refresh `tmp/mondo.sssom.tsv` when needed
This would be cool to do at some point. Basically, in addition to the change I described above, we'd have an if statement at the top of the goal for `tmp/mondo.sssom.tsv` or `tmp/mondo_repo_built`. It would compare the latest git commit hashes on `master` for  the cloned `mondo` dir in `tmp/` and the actual repo on GitHub. If they're the same, do nothing. Else, it will re-run `tmp/mondo_repo_built` --> `tmp/mondo.sssom.tsv`.

---

I think this merits its own GH issue.

File: src/ontology/slurp/ordo.tsv
Comment (joeflack4) [inline src/ontology/slurp/ordo.tsv:3]: ## Output
@sabrinatoro Here you are! Check out this [google sheet](https://docs.google.com/spreadsheets/d/13styiaK6DhF03VF3IEGg-78sJH1pICOfPWzG5iowUpY/edit#gid=1585402401).

File: src/scripts/ordo_subsets.py
Comment (joeflack4) [inline src/scripts/ordo_subsets.py:21]: ## `get_formatted_subsets_df()`
In `ordo_subsets.py`, which is part of #510, there existed some code I needed to re-use; namely, the reading and formatting of `tmp/ordo-subsets.tsv`. So I created this function for that, which is now used in 2 places. That's what the whole diff in `ordo_subsets.py` is about, just moving some code into this new function.

File: src/ontology/slurp/ordo.tsv
Comment (joeflack4) [reply src/ontology/slurp/ordo.tsv:3]: @sabrinatoro You don't need this merged into `main` any time soon, do you? Or, phrased another way--do you foresee needing this information to show up in another build very soon?

If not, then I suppose that the google sheet will be fine for your immediate purposes. I only bring this up because I was planning to merge this into `develop`, which is our typical workflow,  but I don't know when the next time is that `develop` would get merged into `main`. If you foresee that you'll need to see this information again in a build very soon, then I think we can do a merge of `develop` into `main` after this PR is merged.

Comment (joeflack4) [reply src/ontology/slurp/ordo.tsv:3]: @sabrinatoro Does the `subset` column look good to you?
It will show one of:
- disorder
- group of disorders
- subtype of a disorder

If you want, alternatively I could have it show one of these things instead:
```python
    'Orphanet:557493': 'http://purl.obolibrary.org/obo/mondo#ordo_disease',  # disorder
    'Orphanet:557492': 'http://purl.obolibrary.org/obo/mondo#ordo_group_of_disorders',  # group of disorders
    'Orphanet:557494': 'http://purl.obolibrary.org/obo/mondo#ordo_clinical_subtype',  # subtype of a disorder
```

File: src/ontology/mondo-ingest.Makefile
Comment (twhetzel) [reply src/ontology/mondo-ingest.Makefile]: I agree this could use it's own GH issue and maybe a ~todo/TODO for the line in this PR to evaluate the outcome of the future ticket

File: src/ontology/slurp/ordo.tsv
Comment (twhetzel) [reply src/ontology/slurp/ordo.tsv:3]: I also took a look at the spreadsheet and things look good

Joe, if in the future you're interested to know more about Orphanet for things like this, one place to check is the Orphanet web site (https://www.orpha.net/en/disease) and search for a disease, e.g. saccular spinal dysraphism with a stalk to the dome. When you click on the resulting [link](https://www.orpha.net/en/disease/detail/645319?name=saccular%20spinal%20dysraphism%20with%20a%20stalk%20to%20the%20dome&mode=name) for the disease name and scroll down the Classification level is listed on the page.

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [reply src/ontology/mondo-ingest.Makefile]: @twhetzel Sounds good! I created:
- #538 

However, I didn't add a `todo` because this is more pervasive; there are 4 goals that use `tmp/mondo.sssom.tsv` as a dependency. If you want I can insert a `todo` on each of them.

Comment (twhetzel) [reply src/ontology/mondo-ingest.Makefile]: RE: `todo` - whichever works better for your tracking

**Comment by sabrinatoro** [comment]
Thank you @joeflack4 ! That looks great! The additional column in the spreadsheet is exactly what we needed, and it will make review easier. Thank you! 

I do not need this merged into main soon. So there is no urgency from my perspective.

File: src/ontology/slurp/ordo.tsv
Comment (joeflack4) [reply src/ontology/slurp/ordo.tsv:3]: [Sabrina approves](https://github.com/monarch-initiative/mondo-ingest/pull/531#issuecomment-2128148710)! Marking this one resolved.

**Review by twhetzel — APPROVED** [review]
Looks good to me, just one comment for clarification

File: src/scripts/migrate.py
Comment (joeflack4) [reply src/scripts/migrate.py:189]: Good question! Basically we're editing `slurp/ordo.tsv` to add a `subset` column. The edge case is, if `slurp/ordo.tsv` exists already and it already has that column, we want to replace whatever it has there. There are a couple ways to do this, but I simply deleted the column from the slurp file, then joined with the separate file which lists all ORDO classes and their subsets, which reside in a column called 'subset'. So if I didn't delete it first, I'd have 2 copies of subset columns after the join, subset_x and subset_y I think, and then I'd have to rename one of them and drop the other. So the more elegant way seemed to be simply deleting the column that was going to be overwritten anyway.

**Pull request merged**
